cmd/context/create.go

```diff
@@ -133,13 +137,15 @@ func (c *ContextCommand) UseContext(ctx context.Context, ctxOptions *ContextOpti
 			transformedCtx := okteto.K8sContextToOktetoUrl(ctx, ctxOptions.Context, ctxOptions.Namespace, c.K8sClientProvider)
 			if transformedCtx != ctxOptions.Context {
 				ctxOptions.Context = transformedCtx
+				ctxOptions.Namespace = ctxStore.Contexts[ctxOptions.Context].Namespace
 				ctxOptions.IsOkteto = true
 			}
 		}
 	}
 
 	if okCtx, ok := ctxStore.Contexts[ctxOptions.Context]; !ok {
```

Comment:
I'd move all the repeated code to this if else statement. It will be easier to make a change in the future if we only need to check one place